File: priceScrape/src/priceScrape.java
package webscrape;

// import org.apache.poi.hssf.usermodel.HSSFCell;
// import org.apache.poi.hssf.usermodel.HSSFRow;
// import org.apache.poi.hssf.usermodel.HSSFSheet;
// import org.apache.poi.hssf.usermodel.HSSFWorkbook;

import java.util.Scanner;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

public class priceScrape{
	public static void main(String[] args){
		final String walmart = "https://www.walmart.com/ip/Frito-Lay-Fun-Times-Mix-Variety-Pack-40-Count/440311832";
		final String base = "https://www.mygrocerydeals.com/deals?utf8=%E2%9C%93&authenticity_token=KFo1qPeErMj0RspD%2F8AhWJlc9BChi8x8L6irKF9pfCDTL4%2B7TkK0EgkeC85l933qffsBAuiYEHhOFgqQUy0nJg%3D%3D&remove%5B%5D=chains&remove%5B%5D=categories&remove%5B%5D=collection&remove%5B%5D=collection_id&q=";
		final String ending = "&supplied_location=93107&latitude=34.42&longitude=-119.69999999999999";
		// final String ending = "&supplied_location=";
		
		Scanner input1 = new Scanner(System.in);
		String product = input1.nextLine();
		product = product.replaceAll(" ", "+");

		// Scanner input2 = new Scanner(System.in);
		// String zipCode = input2.nextLine();

		// HSSFWorkbook workbook = new HSSFWorkbook();
		// HSSFSheet sheet = workbook.createSheet("Sales");
		
		try{
			final Document document = Jsoup.connect(base+product+ending).get();
			int count = 0;
			for (Element item : document.select("div.container-tile.deal.paged.special-tile")){
				String price = item.select("span.pricetag").text();
				String productName = item.select("p.deal-productname").text();
				String storeName = item.select("p.deal-storename").text();
				String dealEnd = item.select("div.expirydate").text();

				// HSSFRow row = sheet.createRow(count);
				// HSSFCell cell1 = row.createCell(0);
    //             HSSFCell cell2 = row.createCell(1);
    //             HSSFCell cell3 = row.createCell(2);
    //             HSSFCell cell4 = row.createCell(3);
                
    //             cell1.setCellValue(productName);
    //             cell2.setCellValue(price);
    //             cell3.setCellValue(dealEnd);
    //             cell4.setCellValue(storeName);
    //             cout++;

				System.out.println("Product Name: " + productName);
				System.out.println("Price: " + price);
				System.out.println(dealEnd);
				System.out.println("Store: " + storeName + "\n");
			}
		} catch (Exception e) {
			e.printStackTrace();
		}

		// workbook.write(new FileOutputStream("deals.xls")); //title what you want the excel file to be called
        // workbook.close();

		/*try{
			final Document document = Jsoup.connect(walmart).get();
			final String newPrice = document.select("span.hide-content-m").text();
			final String originalPrice = document.select("div.display-inline-block.product-secondary-price.Price-enhanced").text();
			System.out.println("Original Price: "+ originalPrice);
			System.out.println("New Price: "+ newPrice);
		} catch (Exception e) {
			e.printStackTrace();
		}*/

	}
}

File: priceScrape/priceScrape.py

from urllib.request import urlopen as ureq 	#use this one if you're running it as python3 priceScrape.py
from bs4 import BeautifulSoup as soup

groceryUrlBeginning = "https://www.mygrocerydeals.com/deals?utf8=%E2%9C%93&authenticity_token=KFo1qPeErMj0RspD%2F8AhWJlc9BChi8x8L6irKF9pfCDTL4%2B7TkK0EgkeC85l933qffsBAuiYEHhOFgqQUy0nJg%3D%3D&remove%5B%5D=chains&remove%5B%5D=categories&remove%5B%5D=collection&remove%5B%5D=collection_id&q="
groceryUrlEnd = "&supplied_location=93107&latitude=34.42&longitude=-119.69999999999999"
walmartUrlBeginning = "https://www.walmart.com/search/?cat_id=0&query="
targetUrlBeginning = "https://www.target.com/s?searchTerm="
amazonUrlBeginning = "https://www.amazon.com/s?k="


product = input("Please enter the product: ")
fileName = product.title().replace(" ", "") + ".csv"
product = product.replace(" ", "+")

# upci = input("UPCI: ")


f = open(fileName, "w")						#opens file for writing
f.write("Product Name" + ";" + "Size/Count" + ";" + "Price" + ";" + "Deal End" + ";" + "Store Name" + "\n") #setting headers of csv file

############################################################Searches mygrocerydeals.com############################################################


url = groceryUrlBeginning + product + groceryUrlEnd 	#page URL

uclient = ureq(url) 					#opening connection to website	
pageHTMl = uclient.read()				#reading HTML
uclient.close()
content = soup(pageHTMl, "html.parser")

#grabs each product
itemContainers = content.findAll("div", {"data-type":"special"})


count = 0 																			#counter for going row by row
for container in itemContainers:
	productNameContainer = container.findAll("p", {"class":"deal-productname"})		#Finding product name
	productName = productNameContainer[0].text

	sizeContainer = container.findAll("div", {"class":"uom"})						#Finding size/count of product
	size = sizeContainer[0].text

	priceContainer = container.findAll("span", {"class":"pricetag"})				#Finding Price
	price = priceContainer[0].text

	dealEndContainer = container.findAll("div", {"class":"expirydate"})				#Finding deal end date
	dealEnd = dealEndContainer[0].text

	storeNameContainer = container.findAll("p", {"class":"deal-storename"})			#Finding store of sale
	storeName = storeNameContainer[0].text

	f.write(productName + ";" + size + ";" + price + ";" + dealEnd + ";" + storeName + "\n")
	count+=1

###################################################################################################################################################














###########################################################Searches target.com#############################################################

# url = targetUrlBeginning + upci 	#page URL
# url = "https://www.target.com/s?searchTerm=Lay%27s+Classic+Potato+Chips+-+8oz"

# uclient = ureq(url) 					#opening connection to website	
# pageHTMl = uclient.read()				#reading HTML
# uclient.close()
# content = soup(pageHTMl, "html.parser")

# priceContainer = content.findAll("ul", {"data-test":"product-price"})
# if(priceContainer):
# 	price = priceContainer[0].text
# 	print(price)
# else:
# 	print(priceContainer)

f.close() #closes off the file
